ray-curator/ray_curator/backends/experimental/ray_data/adapter.py

```diff
@@ -17,8 +19,9 @@ class RayDataStageAdapter(BaseStageAdapter):
     This adapter converts stages to work with Ray Data datasets by:
     1. Working directly with Task objects (no dictionary conversion)
     2. Using Ray Data's map_batches for parallel processing
-    3. Handling single and batch processing modes
-    4. TODO: Supporting setup() and setup_on_node() calls like other backends
+        a. If stage has both gpus and cpus specified, then we use actors
+        b. If stage.setup is overridden, then we use actors
+        c. Else we use tasks
```

Comment:
Will we want make this overridable (e.g., with a parameter) in the future? Any chance it could depend on the stage's modality?